### PR TITLE
skip uniqueness check on empty codes

### PR DESCRIPTION
**Pull request by sravfeyn**
http://manage.dimagi.com/default.asp?237940 @esoergel

### CODE REVIEW

File: corehq/apps/locations/static/locations/ko/location_types.js [hunk 1]
```diff
@@ -90,6 +90,11 @@ hqDefine('locations/ko/location_types.js', function(){
                 });
                 var duplicates = [];
                 _.each(counts_by_value, function (count, value) {
+                    if (field == 'code' && value == ''){
```
Comment (dimagimon) [inline corehq/apps/locations/static/locations/ko/location_types.js]: Expected '===' and instead saw '=='. (eqeqeq)

File: corehq/apps/locations/views.py [hunk 2]
```diff
@@ -252,10 +252,12 @@ def mk_loctype(name, parent_type, administrative,
...
-        if not (
-            len(loc_types) == len(set(lt['name'] for lt in loc_types))
-            and len(loc_types) == len(set(lt['code'] for lt in loc_types))
-        ):
+
+        names = [lt['name'] for lt in loc_types]
+        unique_names = len(names) == len(set(names))
+        codes = [lt['code'] for lt in loc_types if lt['code']]
+        unique_codes = len(codes) == len(set(codes))
+        if not unique_names or not unique_codes:
```
Comment (esoergel) [inline corehq/apps/locations/views.py]: boolean names might be clearer here - `unique_codes` sounds like a list of unique codes.  Something like:
```python
has_duplicate_codes = len(codes) != len(set(codes))
```

**Review by esoergel — APPROVED** [review]
lgtm, feel free to ignore my comment if you disagree

File: corehq/apps/locations/static/locations/ko/location_types.js [hunk 3]
```diff
@@ -90,6 +90,11 @@ hqDefine('locations/ko/location_types.js', function(){
                 });
                 var duplicates = [];
                 _.each(counts_by_value, function (count, value) {
+                    if (field === 'code' && value === ''){
+                        // exclude empty codes
+                        // if code is empty, the backend will autofill it as name
```
Comment (sravfeyn) [inline corehq/apps/locations/static/locations/ko/location_types.js]: Yeah I've tested it

File: corehq/apps/locations/views.py (hunk 2 above)
Comment (sravfeyn) [inline corehq/apps/locations/views.py]: (changed it)

Comment (esoergel) [inline corehq/apps/locations/views.py]: thanks!

**Pull request merged**